src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py

```diff
@@ -1007,13 +1016,92 @@ def load_pipeline_from_original_stable_diffusion_ckpt(
             Whether the attention computation should always be upcasted. This is necessary when running stable
             diffusion 2.1.
         device (`str`, *optional*, defaults to `None`):
-            The device to use. Pass `None` to determine automatically. :param from_safetensors: If `checkpoint_path` is
-            in `safetensors` format, load checkpoint with safetensors instead of PyTorch. :return: A
-            StableDiffusionPipeline object representing the passed-in `.ckpt`/`.safetensors` file.
+            The device to use. Pass `None` to determine automatically.
+        from_safetensors (`str`, *optional*, defaults to `None`):
+            If `checkpoint_path` is in `safetensors` format, load checkpoint with safetensors instead of PyTorch.
+            Pass `None` to determine automatically based on checkpoint file extension
+        pipeline_class (`str`, *optional*, defaults to `None`):
+            The pipeline class to use. Pass `None` to determine automatically.
+
+        
     """
+
+    # import pipelines here to avoid circular import error when using from_ckpt method
+    from diffusers import (
+        StableDiffusionControlNetPipeline,
+        StableDiffusionPipeline,
+        StableUnCLIPImg2ImgPipeline,
+        StableUnCLIPPipeline,
+        LDMTextToImagePipeline,
+        PaintByExamplePipeline
+    )
+
+    if pipeline_class:
+
+        pipeline_name = pipeline_class.__name__
+
+        if pipeline_name == "StableDiffusionControlNetPipeline":
+            model_type = "FrozenCLIPEmbedder"
+            controlnet = True
+
+        elif "StableDiffusion" in pipeline_name:
+            model_type = "FrozenCLIPEmbedder"
+            controlnet = False
+
+        elif pipeline_name == "StableUnCLIPImg2ImgPipeline":
+            model_type == "FrozenOpenCLIPEmbedder"
+            stable_unclip = "img2img"
+
+        elif pipeline_name == "StableUnCLIPPipeline":
+            model_type == "FrozenOpenCLIPEmbedder"
+            stable_unclip = "txt2img"
+
+        elif pipeline_name == "PaintByExamplePipeline":
+            model_type == "PaintByExample"
+
+        elif pipeline_name == "LDMTextToImagePipeline":
+            model_type == "LDMTextToImage"
+
+        else:
+            raise ValueError(f"Unhandled pipeline class: {pipeline_name}")
+
+    else:
+        pipeline_class = StableDiffusionPipeline
```

Comment:
I'm not really a fan of this. If the model_type and other parameters have to be set, they should just be passed in as arguments and not overridden by the pipeline.

If the goal is just to support pipeline classes other than the default ones used by the script, just pass in the class and use it in lieu of the other constructor.

i.e.
```py
if pipeline_class is not None:
    pipeline_class(<args>)
else:
   StableDiffusionPipeline(<args>) # <- or other pipeline that is used by default
```

Reply:
Yes good point, it is duplicative with the `model_type` parameter. I've moved this logic down to FromCkptMixin as well